extensions/ql-vscode/src/remote-queries/analyses-results-manager.ts

```diff
@@ -0,0 +1,90 @@
+import { ExtensionContext } from 'vscode';
+import { Credentials } from '../authentication';
+import { Logger } from '../logging';
+import { downloadArtifactFromLink } from './gh-actions-api-client';
+import { DownloadLink } from './download-link';
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import { AnalysisSummary } from './shared/remote-query-result';
+import * as sarif from 'sarif';
+import { AnalysisResults, QueryResult } from './shared/analysis-result';
+
+export class AnalysesResultsManager {
+  // Store for the results of various analyses for a single remote query.
+  private readonly analysesResults: { [key: string]: QueryResult[] };
+
+  constructor(
+    private readonly ctx: ExtensionContext,
```

Comment:
Rather than storing the `ctx` as a field in this class, does it make more sense to create a credentials instance in the constructor and use that?

I'm not sure if this is a good idea since credentials instances may expire, so take this as a question only.

Reply:
Yeah I think storing the `ctx` is safer because the creds might expire.